src/software/ai/hl/stp/tactic/receiver/receiver_fsm.cpp

```diff
@@ -156,37 +158,37 @@ void ReceiverFSM::updateReceive(const Update& event)
 
 void ReceiverFSM::adjustReceive(const Update& event)
 {
-    auto ball      = event.common.world_ptr->ball();
-    auto robot_pos = event.common.robot.position();
+    const Ball& ball   = event.common.world_ptr->ball();
+    const Robot& robot = event.common.robot;
 
-    if ((ball.position() - robot_pos).length() >
-        BALL_TO_FRONT_OF_ROBOT_DISTANCE_WHEN_DRIBBLING)
-    {
-        Point ball_receive_pos = ball.position();
-
-        if (ball.velocity().length() > MIN_PASS_START_SPEED)
-        {
-            ball_receive_pos = closestPoint(
-                robot_pos, Line(ball.position(), ball.position() + ball.velocity()));
-        }
-
-        Angle ball_receive_orientation = (ball.position() - robot_pos).orientation();
+    auto face_ball_orientation = (ball.position() - robot.position()).orientation();
 
-        event.common.set_primitive(std::make_unique<MovePrimitive>(
-            event.common.robot, ball_receive_pos, ball_receive_orientation,
-            TbotsProto::MaxAllowedSpeedMode::PHYSICAL_LIMIT,
-            TbotsProto::ObstacleAvoidanceMode::AGGRESSIVE,
-            TbotsProto::DribblerMode::MAX_FORCE, TbotsProto::BallCollisionType::ALLOW,
-            AutoChipOrKick{AutoChipOrKickMode::OFF, 0}));
+    Point intercept_position;
+    if (strayPass(event) || ball.velocity().length() <= MIN_STRAY_PASS_SPEED)
+    {
+        intercept_position =
+            findInterceptionPoint(robot, ball, event.common.world_ptr->field());
     }
+    else
+    {
+        intercept_position = closestPoint(
+            robot.position(), Line(ball.position(), ball.position() + ball.velocity()));
+    }
+
+    event.common.set_primitive(std::make_unique<MovePrimitive>(
+        event.common.robot, intercept_position, face_ball_orientation,
+        TbotsProto::MaxAllowedSpeedMode::PHYSICAL_LIMIT,
+        TbotsProto::ObstacleAvoidanceMode::AGGRESSIVE,
```

Comment:
i'd argue we don't need AGGRESSIVE obstacle avoidance here

Reply:
I think AGGRESSIVE is good bc we want to get the ball. If we were SAFE then we might take some longwinded path just to avoid other robots